app/controllers/arts_controller.rb

```diff
@@ -0,0 +1,33 @@
+class ArtsController < ApplicationController
+  before_action :set_user, only: %i[new create]
+
+  def index
+    Art.all
+  end
+
+  def new
+    Art.new
+  end
+
+  def create
+    @art = Art.new(art_params)
+    @art.user = @user
+    if Art.save
+      # unsure if this is correct
+      redirect_to user_path(@user)
+    else
+      render 'new', status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def set_user
+    @user = User.find(params[:user_id])
+  end
```

Comment:
The Art routes are not nested in the User's so you can't  set_user using params.
But you don't need it either because there is a helper method coming with Devise (the authentification gem) that allows you to write `current_user` to get the current user.
That means you can remove the `set_user` method, the `before_action :set_user` callback, then inside `create` you set `@art.user = current_user`
Because we use the current_user that also means we don't need a @user instance variable for the `new` action

Reply:
And regarding the redirect, you should `redirect_to art_path(@art)` instead

Reply:
done!